legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/nodes/helpers/freemarker/FreeMarkerExecutor.java

```diff
@@ -66,12 +67,19 @@ public static String process(String input, ExecutionState executionState, String
 
     public static String processRecursively(String input, Map<String, ?> variableMap, String templateFunctions)
     {
-        String result = process(input, variableMap, templateFunctions);
-        if (!result.equals(input.replace("\\\"", "\"")))
+        return processRecursivelyInternal(input, variableMap.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)), templateFunctions);
```

Comment:
Could you extract out the second argument (the deep copy of variableMap) into a variable for readability?
Also might be worth adding a comment to say why you need to do a deep copy of variableMap